lib/puppet/provider/service/windows.rb

```diff
@@ -125,4 +124,18 @@ def self.instances
     end
     services
   end
+
+  def logonaccount
+    return '[Unknown]' unless Puppet::Util::Windows::Service.exists?(@resource[:name])
```

Comment:
This currently results in:
> service { 'non-existing-service-name':
  ensure       => 'stopped',
  enable       => 'false',
  logonaccount => '[Unknown]',
  provider     => 'windows',
}

Just chose something with characters that cannot be found in usernames but it can be changed if there are any concerns.

Reply:
I would just return `nil` here, then `logonaccount` won't show up at all when calling `puppet resource` on a unexisting service.

Reply:
Agreed, I think that would be better.